2_process.smk

```diff
@@ -0,0 +1,155 @@
+import os
+import numpy as np
+import pandas as pd
+
+configfile: "2_process/process_config.yaml"
+
+
+def mntoha_lake_sequence_files(wildcards):
+    """
+    List all MNTOHA lake sequence .npy files for training and testing.
+
+    :param wildcards: Snakemake wildcards (empty).
+    :returns: List of lake training/testing sequence files.
+
+    """
+    # make this function dependent on fetch_mntoha_metadata
+    # needed because lake_metadata.csv is used to determine lake_sequence_files
+    lake_metadata_file = checkpoints.fetch_mntoha_metadata.get(**wildcards).output[0]
+    lake_metadata = pd.read_csv(lake_metadata_file)
+    lake_sequence_files = [os.path.join('2_process/out/mntoha_sequences', f'sequences_{site_id}.npy') for site_id in lake_metadata.site_id]
```

Comment:
Hmm one thing that doesn't seem ideal is that if you changed the filename template on line 108, it wouldn't match what is defined here 🤔

Just thinking out loud here -- I wonder if it would be more robust to do this: 

switch line 94 from ` mntoha_lake_sequence_files` to
```python
lambda wildcards: mntoha_lake_sequence_files(sequence_file_template='2_process/out/mntoha_sequences/sequences_{}.npy')
```

and then switch this line to
``` python
lake_sequence_files = [sequence_file_template.format(site) for site_id in lake_metadata.site_id]
```
and add `sequence_file_template` as an argument to `mntoha_lake_sequence_files()`

That would pull the file template down into the 'rules' section, so would make it a little less hidden, though it still would need to be updated manually if line 108 changed. 🤷‍♀️ What do you think?

Reply:
Yes, keeping the filename template in the rules and not hardcoded in functions would be a more robust solution. I think that will set us up better for the future, too, when we incorporate data beyond MNTOHA.